tests/semantic_analyser.cpp

```diff
@@ -2349,6 +2349,80 @@ TEST(semantic_analyser, map_cast_types)
 
 TEST(semantic_analyser, map_aggregations_implicit_cast)
 {
+  // When assigning an aggregation to a map containing integers,
+  // the aggregation is implicitly cast to an integer.
+  test("kprobe:f { @x = 1; @y = count(); @x = @y; }", R"(*
+  =
+   map: @x :: [int64]
+   (int64)
+    map: @y :: [count_t]
+*)");
+  test("kprobe:f { @x = 1; @y = sum(5); @x = @y; }", R"(*
+  =
+   map: @x :: [int64]
+   (int64)
+    map: @y :: [sum_t]
+*)");
+  test("kprobe:f { @x = 1; @y = min(5); @x = @y; }", R"(*
+  =
+   map: @x :: [int64]
+   (int64)
+    map: @y :: [min_t]
+*)");
+  test("kprobe:f { @x = 1; @y = max(5); @x = @y; }", R"(*
+  =
+   map: @x :: [int64]
+   (int64)
+    map: @y :: [max_t]
+*)");
+  test("kprobe:f { @x = 1; @y = avg(5); @x = @y; }", R"(*
+  =
+   map: @x :: [int64]
+   (int64)
+    map: @y :: [avg_t]
+*)");
+
+  // Assigning to a newly declared map requires an explicit cast
+  // to get the value of the aggregation.
+  test("kprobe:f { @x = count(); @y = (uint64)@x; }");
+  test("kprobe:f { @x = sum(5); @y = (uint64)@x; }");
+  test("kprobe:f { @x = min(5); @y = (uint64)@x; }");
+  test("kprobe:f { @x = max(5); @y = (uint64)@x; }");
+  test("kprobe:f { @x = avg(5); @y = (uint64)@x; }");
+
+  // However, if there is no explicit cast,
+  // the assignment is rejected and casting is suggested.
+  test_error("kprobe:f { @y = count(); @x = @y; }", R"(
+stdin:1:26-33: ERROR: Map value 'count_t' cannot be assigned from one map to another. The function that returns this type must be called directly e.g. `@x = count();`.
+Add a cast to integer if you want the value of the aggregate.
```

Comment:
I feel like this hint should go after the error, perhaps with a small example of the cast.

Reply:
Is that something we can do with our current `LOG` facilities?

Reply:
Try `LOG(HINT, cast.loc, err_) `